### PR TITLE
add implementations of setTopic and sayResponse

### PR DESCRIPTION
**Pull request by jfmherokiller**

### CODE REVIEW

File: include/RE/M/MenuTopicManager.h [hunk 1]
```diff
@@ -55,11 +56,10 @@ namespace RE
...
 
-		static MenuTopicManager* GetSingleton()
-		{
-			REL::Relocation<MenuTopicManager**> singleton{ REL::ID(514959) };
-			return *singleton;
-		}
+		static MenuTopicManager* GetSingleton();
+
+		void SayResponse(std::uint32_t a_chosenIndex, void* a_arg2);
+		bool SetTopic(std::uint32_t ChosenIndex);
```
Comment (Qudix) [inline include/RE/M/MenuTopicManager.h:62]:
```suggestion
		bool SetTopic(std::uint32_t a_chosenIndex);
```

File: include/RE/M/MenuTopicManager.h [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include "RE/B/BSTEvent.h"
 #include "RE/B/BSTList.h"
 #include "RE/B/BSTSingleton.h"
+#include "RE/D/DialogueItem.h"
```
Comment (Qudix) [inline include/RE/M/MenuTopicManager.h:9]:
```suggestion
```
Is this needed?

Comment (jfmherokiller) [reply include/RE/M/MenuTopicManager.h:9]: if i avoid creating a cpp file its not but if a cpp file is used then yes

Comment (jfmherokiller) [reply include/RE/M/MenuTopicManager.h:9]: essentially DialogueResponse becomes undefined

**Comment by Ryan-rsm-McKenzie** [comment]
Please add spaces between these functions, there's no global shortage of whitespace.

File: include/RE/M/MenuTopicManager.h [hunk 3]
```diff
@@ -61,6 +61,19 @@ namespace RE
...
+			return func(this, a_chosenIndex);
+		}
+
+		void SayResponse(std::uint32_t a_chosenIndex, void* a_arg2 = nullptr)
+		{
+			using func_t = decltype(&MenuTopicManager::SayResponse);
+			REL::Relocation<func_t> func{ REL::ID(34456) };
+			return func(this, a_chosenIndex, a_arg2);
+		}
 		// members
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/M/MenuTopicManager.h:77]:
```suggestion
		}
		
		// members
```

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

**Pull request closed without merging**